app/src/main/java/com/example/concordia_campus_guide/Adapters/RoutesAdapter.java

```diff
@@ -0,0 +1,208 @@
+    public int getCount(){
+        return routes.size();
+    }
+
+    // Helper methods for setting UI elements in the Routes Overview section of the Routes Activity page
+    private void setUIElements(Route route) {
+        if (route.getMainTransportType().equals(ClassConstants.WALKING)) { setUIWalk(route); }
+
+        else if (route.getMainTransportType().equals(ClassConstants.DRIVING)) { setUIDriving(route); }
+
+        else { // TRANSIT
+            setUITransit(route);
+
+            // Go through each steps in the route
+            for(int i = 0; i < route.getSteps().size(); i++) {
+                TransportType step = route.getSteps().get(i);
+
+                if (step instanceof Bus) { setUIBus((Bus) step); }
+                else if (step instanceof Subway) { setUISubway((Subway) step); }
+                else if (step instanceof Train) { setUITrain((Train) step); }
+                else if (step instanceof Walk) { setUIWalkInTransit((Walk) step); }
+
+                if(i < route.getSteps().size() - 1) { setUIArrow(); }
+            }
+        }
+
+        setBottomLayout(route);
+    }
+    private void setUIWalk(Route route) {
+        mainTransportType.setImageResource(R.drawable.ic_directions_walk_red);
+
+        TextView summary = new TextView(context);
+        summary.setText(route.getSummary());
+        summary.setGravity(Gravity.CENTER);
+        details.addView(summary);
+
+        duration.setText(route.getDuration());
+    }
+
+    private void setUIWalkInTransit(Walk walk) {
+        ImageView walkIcon = new ImageView(context);
+        walkIcon.setImageResource(R.drawable.ic_directions_walk_red);
+
+        TextView durationWalk = new TextView(context);
+        durationWalk.setText(walk.getDuration());
+        durationWalk.setGravity(Gravity.CENTER);
+
+        details.addView(walkIcon);
+        details.addView(durationWalk);
+    }
+
+    private void setUIDriving(Route route) {
+        mainTransportType.setImageResource(R.drawable.ic_directions_car_red);
+
+        TextView summary = new TextView(context);
+        summary.setText(route.getSummary());
+        summary.setGravity(Gravity.CENTER);
+        details.addView(summary);
+
+        duration.setText(route.getDuration());
+    }
+
+    private void setUIBus(Bus bus) {
+        ImageView busIcon = new ImageView(context);
+        busIcon.setImageResource(R.drawable.ic_directions_bus_red);
+
+        TextView busNumber = new TextView(context);
+        busNumber.setText(bus.getBusNumber());
+        busNumber.setGravity(Gravity.CENTER);
+
+        details.addView(busIcon);
+        details.addView(busNumber);
+    }
+
+    private void setUISubway(Subway subway) {
+        ImageView subwayIcon = new ImageView(context);
+        subwayIcon.setImageResource(pickCorrectSubwayColor(subway.getColor()));
+
+        details.addView(subwayIcon);
+    }
+
+    private int pickCorrectSubwayColor(String color) {
+        if(color.equalsIgnoreCase("#0074b9"))
```

Comment:
I think it is possible to reference colors from Colors.xml here instead of hex code, it will be more reusable

Reply:
like get the color from Color.xml and get the hex value of it to compare them instead of harcoding the hex?

Reply:
makes perfect sense, will do that!